source/extensions/stat_sinks/hystrix/hystrix.cc

```diff
@@ -304,6 +321,32 @@ void HystrixSink::flush(Stats::Source&) {
   incCounter();
   std::stringstream ss;
   Upstream::ClusterManager::ClusterInfoMap clusters = server_.clusterManager().clusters();
+
+  // Save a map of the relevant histograms per cluster in a convenient format.
+  std::unordered_map<absl::string_view, std::unordered_map<std::string, double>, StringViewHash>
+      time_histograms;
+  for (const Stats::ParentHistogramSharedPtr histogram : server_.stats().histograms()) {
+    // histogram->name() on clusters of the format "cluster.cluster_name.histogram_name"
+    // i.e. "cluster.service1.upstream_rq_time".
+    const std::vector<absl::string_view> split_name = absl::StrSplit(histogram->name(), '.');
+    if (split_name[0] == "cluster" && split_name[2] == "upstream_rq_time") {
+      std::unordered_map<std::string, double>& hist_map = time_histograms[split_name[1]];
+      for (size_t i = 0; i < histogram->cumulativeStatistics().supportedQuantiles().size(); ++i) {
+        if (std::find(hystrix_quantiles.begin(), hystrix_quantiles.end(),
+                      histogram->cumulativeStatistics().supportedQuantiles()[i]) !=
```

Comment:
Can you make a temp ref for supportedQuartiles()? Would make this long std::find call easier to read.